src/main/java/Parser.java

```diff
@@ -0,0 +1,98 @@
+/** Parser class Responsible for parsing user commands */
+public class Parser {
+
+  /**
+   * Takes in user commands in a tokenized string form Updates tasklist accordingly
+   *
+   * @param inputs User inputs
+   * @param tasks TaskList
+   */
+  public void parseCommand(String[] inputs, TaskList tasks) {
+    String command = inputs[0];
+    switch (command) {
+      case "list":
+        tasks.list();
+        break;
+      case "done":
+        try {
+          tasks.handleDone(inputs);
+        } catch (DukeException e) {
+          System.out.println(e);
```

Comment:
I noticed that the try-catch block is being duplicated in quite a few cases. Perhaps you can consider moving the moving the try-catch block one level up, putting the switch case in the try block and catch DukeException?